rocketpy/Function.py

```diff
@@ -600,8 +636,8 @@ def setDiscreteBasedOnModel(self, modelFunction, oneByOne=True):
             Zs = np.array(self.getValue(mesh))
             self.setSource(np.concatenate(([Xs], [Ys], [Zs])).transpose())
 
-        self.setInterpolation(modelFunction.__interpolation__)
-        self.setExtrapolation(modelFunction.__extrapolation__)
+        self.setInterpolation(self.__interpolation__)
+        self.setExtrapolation(self.__extrapolation__)
         return self
```

Comment:
I prefer keeping the `self` interpolation and extrapolation but maybe leaving an optional argument to use the `modelFunction` inter and extrapolation might be useful in the future

Reply:
Added on [df897a1](https://github.com/RocketPy-Team/RocketPy/pull/354/commits/df897a1670ed6a9e0d543fee4f9fa4f0aac3f748). Thanks for the suggestion.